Replicator/Replicator.cc

```diff
@@ -147,6 +147,20 @@ namespace litecore { namespace repl {
         _disconnect(websocket::kCodeNormal, {});
     }
 
+
+    void Replicator::terminate() {
+        auto conn = connection();
+        if (conn) {
```

Comment:
Just to confirm:  This is a safe assumption right?  Either we've never started and the connection is still closed, or we've already stopped and there is no connection anymore.  There are no other scenarios?

Reply:
Well, it's a precondition (see the header comment.) We know this condition is true in the C4Replicator destructor, because C4Replicator retains itself while the replicator is running, so once it's been started it can't be destructed until after the Replicator has stopped.